tmitchell/exercise.py

```diff
@@ -0,0 +1,281 @@
+    def __init__(self):
+        self.name = "phonebook"
+        self.people = []
+
+    def display_self(self):
+        """Iterate through all entries in book and display details."""
+        print("")
+        print("---Fonbuk 5000 User List---")
+        print("")
+        for i in self.people:
+            i.display()
+            print("")
+        print("---End of Fonbuk---")
+        print("")
+
+    def search(self):
+        """Request criteria from user, iterate through all entries and display matching results."""
+        print("Enter part or whole of entry name:")
+        search = input("")
+        count = 0
+        for i in self.people:
+            if search in i.name:
+                i.display()
+                count = 1
+        if count == 0:
+            print("Could not find any entries matching that description")
+
+    def man_add_person(self):
+        """Request input from user, create new entry based on that then add to phonebook."""
+        print("Enter new entry's name:")
+        new_name = input("")
+        if self.check_duplicate(new_name) == 0:
+            print("An entry with that name already exists,")
+            print("please try again with a different name")
+        else:
+            print("Enter new entry's home number:")
+            new_hnumber = input("")
+            print("Enter new entry's mobile number:")
+            new_mnumber = input("")
+            new_person = Person(new_name, new_hnumber, new_mnumber)
+            self.people.append(new_person)
+
+    def delete_person(self):
+        """Request input from user, identify then delete appropriate entry."""
+        print("Enter exact name of entry to be deleted")
+        delrow = input("")
+        index = self.identify_by_name(delrow)
+        if index == "null":
+            print("Delete failed! Returning to main menu")
+        else:
+            print("Deleting %s" % (self.people[index].name))
+            del self.people[index]
+
+    def edit_person(self):
+        """Request input from user, identify entry and then edit it based on further input."""
+        print("Enter exact name of entry to be edited")
+        edited = input("")
+        index = self.identify_by_name(edited)
+        if index == "null":
+            print("Edit failed! Returning to main menu")
+        else:
+            print("Current entry Name: %s" % (self.people[index].name))
+            print("Enter new name: (leave blank to remain the same)")
+            new_name = input("")
+            dupe = self.check_duplicate(new_name)
+            if (dupe == 0) & (new_name != self.people[index].name):
+                print("An entry with that name already exists,")
+                print("please try again with a different name")
+            else:
+                self.people[index].edit(new_name)
+
+    def identify_by_name(self, name):
+        """Identify entry then return entry's position in book."""
+        count = 0
+        for i in self.people:
```

Comment:
if you do `for idx, person in enumerate(self.people):` then you have access to the index without using count.